backend/users/serializers.py

```
@@ -120,28 +124,48 @@ class PasswordResetConfirmSerializer(serializers.Serializer):
-            raise serializers.ValidationError({"token": "Invalid or expired token."})
-
         try:
             validate_password(password, user=user)
         except DjangoValidationError as e:
+            record_password_reset_event(
+                user,
+                "failed",
+                "failure",
+                metadata,
+            )
             raise serializers.ValidationError({"password": list(e.messages)})
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)